server/src/main/java/io/crate/expression/udf/UserDefinedFunctionService.java

```diff
@@ -240,4 +266,49 @@ public FunctionProvider buildFunctionResolver(UserDefinedFunctionMetadata udf) {
 
         return new FunctionProvider(signature, (s, args) -> scalar);
     }
+
+    private void validateFunctionIsNotInUseByGeneratedColumn(String schema,
+                                                             String functionName,
+                                                             UserDefinedFunctionsMetadata functionsMetadata,
+                                                             ClusterState currentState) {
+        // The iteration of schemas/tables must happen on the node context WITHOUT the UDF already removed.
+        // Otherwise the lazy table factories will already fail while evaluating generated functionsMetadata.
+        // To avoid that, a copy of the node context with the removed UDF function is used on concrete expression evaluation.
+        var nodeCtxWithRemovedFunction = new NodeContext(this.nodeCtx.functions().copyOf());
+        updateImplementations(schema, functionsMetadata.functionsMetadata().stream(), nodeCtxWithRemovedFunction);
+
+        var metadata = currentState.metadata();
+        var indices = Stream.of(metadata.getConcreteAllIndices()).filter(NO_BLOB_NOR_DANGLING)
+            .map(IndexParts::new)
+            .filter(indexParts -> !indexParts.isPartitioned())
```

Comment:
If we filter out all partitions, wouldn't we also need to go through the templates to make sure we also check that the UDF is not used within a partitioned table?

Reply:
damn I forgot 🤦

Reply:
fixup pushed